### PR TITLE
User can edit name and birthday from profile page

### PR DESCRIPTION
**Pull request by dplamenov**
User can edit name and birthday from profile page.

## Motivation and context
User can edit name and birthday from profile page. Issue #608 

## Screenshots:
Screenshot at Apr 09 12-59-20
Screenshot at Apr 09 12-59-46
Screenshot at Apr 09 13-00-01

### CODE REVIEW

File: public/locales/bg/common.json [hunk 1]
```diff
@@ -1,81 +1,82 @@
 {
```
Comment (kachar) [inline public/locales/bg/common.json:1]: Please undo the whitespace change of this file, seems unecessary

File: src/common/util/useCurrentPerson.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  return useQuery<CurrentPerson>(
+    endpoints.account.me.url,
+    authQueryFnFactory<CurrentPerson>(keycloak?.token),
+  )
+}
+
+export function updateCurrentPerson() {
+  const { keycloak } = useKeycloak<KeycloakInstance>()
+  return async (data: UpdatePerson) => {
+    return await apiClient.put<UpdatePerson, AxiosResponse<Person>>(
```
Comment (kachar) [inline src/common/util/useCurrentPerson.ts:25]: Here the response might be:

```suggestion
    return await apiClient.put<UpdatePerson, AxiosResponse<{user: Person}>>(
```

File: src/gql/person.d.ts [hunk 3]
```diff
@@ -43,3 +43,21 @@ export type Person = {
...
+  company?: string
+  createdAt?: Date
+  updatedAt?: Date
+  newsletter?: boolean
+  address?: string
+  birthday?: Date | null
+  emailConfirmed?: boolean
+  personalNumber?: string | null
+  keycloakId?: string | null
+  stripeCustomerId?: string | null
```
Comment (kachar) [inline src/gql/person.d.ts:62]:
```suggestion
  personalNumber?: string | null
```

File: src/gql/person.d.ts [hunk 4]
```diff
@@ -43,3 +43,21 @@ export type Person = {
...
+
+export type UpdatePerson = {
+  id?: UUID
+  firstName?: string
+  lastName?: string
+  email?: string
+  phone?: string
+  company?: string
+  createdAt?: Date
+  updatedAt?: Date
```
Comment (kachar) [inline src/gql/person.d.ts:55]:
```suggestion
```

File: public/locales/bg/common.json (hunk 1 above)
Comment (dplamenov) [reply public/locales/bg/common.json]: Ready.

File: src/gql/person.d.ts [hunk 5]
```diff
@@ -43,3 +43,16 @@ export type Person = {
...
+  firstName?: string
+  lastName?: string
+  email?: string
+  phone?: string
+  company?: string
+  newsletter?: boolean
+  address?: string
+  birthday?: Date | null
+  personalNumber?: string | null
+}
```
Comment (kachar) [inline src/gql/person.d.ts:58]:
```suggestion
export type UpdatePerson = Partial<
  Pick<
    Person,
    | 'id'
    | 'firstName'
    | 'lastName'
    | 'email'
    | 'phone'
    | 'company'
    | 'newsletter'
    | 'address'
    | 'birthday'
    | 'personalNumber'
  >
>

```

**Comment by kachar** [comment]
@dplamenov While testing the PR I've stumbled upon the following error:

1. Open edit name modal
2. Change the name
3. Submit

image


Looks like the API needs a decordator `@IsOptional()` on the `birthday` field and maybe others

**Comment by kachar** [comment]
@dplamenov I've added those decorators myself with https://github.com/podkrepi-bg/api/commit/a1ca5057aa9050bdfc561de294442724b6d715c0

**Comment by dplamenov** [comment]
@kachar Wait 5 min before merge to fix notifications

**Comment by dplamenov** [comment]
@kachar Ready 🚀

**Comment by kachar** [comment]
@dplamenov 

At the moment closing the modal is not possible

image


Please add an X button to close it and maybe a Cancel button to discard the changes

**Comment by dplamenov** [comment]
Screenshot at Apr 10 10-28-03

**Pull request merged**